src/ol/renderer/layerrenderer.js

```diff
@@ -191,7 +195,8 @@ ol.renderer.Layer.prototype.scheduleExpireCache =
              */
             function(tileSource, map, frameState) {
               var tileSourceKey = goog.getUid(tileSource).toString();
-              tileSource.expireCache(frameState.usedTiles[tileSourceKey]);
+              tileSource.expireCache(frameState.viewState.projection,
+                                     frameState.usedTiles[tileSourceKey]);
```

Comment:
The `tileSource.expireCache` call actually expires tiles in all the projections (including the source tiles) if the respective TileCache capacity is over high water mark.
The `projection` parameter only specifies the projection of the `usedTiles` (and these are not removed from the cache even if it's over the high water mark).